src/modules/Utilities/public/Clear-SdnWorkingDirectory.ps1

```diff
@@ -13,34 +18,56 @@ function Clear-SdnWorkingDirectory {
         Forces the cmdlet to remove items that cannot otherwise be changed, such as hidden or read-only files or read-only aliases or variables.
     .EXAMPLE
         PS> Clear-SdnWorkingDirectory
+    .EXAMPLE
+        PS> Clear-SdnWorkingDirectory -ComputerName PREFIX-NC01 -Path 'C:\Temp\SDN2'
+    .EXAMPLE
+        PS> Clear-SdnWorkingDirectory -ComputerName PREFIX-NC01,PREFIX-SLB01 -Credential (Get-Credential)
     .EXAMPLE
         PS> Clear-SdnWorkingDirectory -Force -Recurse
     .EXAMPLE
         PS> Clear-SdnWorkingDirectory -Path 'C:\Temp\SDN1','C:\Temp\SDN2' -Force -Recurse
     #>
 
-    [CmdletBinding()]
+    [CmdletBinding(DefaultParameterSetName = 'Local')]
     param (
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'Remote')]
+        [System.String[]]$ComputerName,
+
+        [Parameter(Mandatory = $false, ParameterSetName = 'Remote')]
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]
+        $Credential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $false, ParameterSetName = 'Remote')]
+        [Parameter(Mandatory = $false, ParameterSetName = 'Local')]
         [System.String[]]$Path = (Get-WorkingDirectory),
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'Remote')]
+        [Parameter(Mandatory = $false, ParameterSetName = 'Local')]
         [Switch]$Recurse,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'Remote')]
+        [Parameter(Mandatory = $false, ParameterSetName = 'Local')]
         [Switch]$Force
     )
 
     try {
-        foreach ($object in $Path) {
-            if ($object -inotlike "$($Global:SdnDiagnostics.Settings.FolderPathsAllowedForCleanup)*") {
-                "{0} is not defined as an allowed path for cleanup. Skipping" -f $object | Trace-Output -Level:Warning
-                continue
+        if ($PSCmdlet.ParameterSetName -eq 'Remote') {
+            Invoke-PSRemoteCommand -ComputerName $ComputerName -Credential $Credential -ScriptBlock {
+                Clear-SdnWorkingDirectory -Path $using:Path -Recurse:($using:Recurse.IsPresent) -Force:($using:Force.IsPresent)
             }
+        }
+        else {
+            foreach ($object in $Path) {
+                if ($object -inotlike "$($Global:SdnDiagnostics.Settings.FolderPathsAllowedForCleanup)") {
```

Comment:
FolderPathsAllowedForCleanup defined as Array of String. This will not compare each of the String in the array to verify if path is allowed cleanup or not. As a result, below warning always reported:
```
WARNING: [SDNEXPMUX01] C:\Windows\Tracing\SdnDataCollection\Temp is not defined as an allowed path for cleanup. 
Skipping
WARNING: [SDNEXPMUX02] C:\Windows\Tracing\SdnDataCollection\Temp is not defined as an allowed path for cleanup. 
Skipping
```

Reply:
Thanks for catching this! I have pushed a new commit that loops through the objects in $Path, as well as the objects defined in  FolderPathsAllowedForCleanup  to compare and if match, then set internal variable to $true which will then be the indicator to remove the path or not.